app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,96 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+)
+
+sealed class MovieCategoryLabelTextViewState{
+    class InputText(val text: String): MovieCategoryLabelTextViewState()
+    class ResourceText(@StringRes val textRes: Int): MovieCategoryLabelTextViewState()
+}
+
+@Composable
+fun MovieCategoryLabel(
+    movieCategoryLabelViewState: MovieCategoryLabelViewState,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    if (movieCategoryLabelViewState.isSelected) {
+        Column(
+            modifier = modifier
+                .padding(5.dp)
+                .width(intrinsicSize = IntrinsicSize.Max)
+                .clickable(onClick = onClick)
```

Comment:
In previous place you wrote `clickable { onClick() }` which is equivalent to `clickable({ onClick() })` and when you name the parameter, you get `clickable(onClick = { onClick() })`. Difference between that and what you wrote here is that in this example there is new lambda created just to call the one which is already there. So we can conclude that `clickable(onClick = onClick) is better because it doesn't introduce unnecessary overhead of creating a new lambda. In practice, that overhead can be ignored, so it is up to you to decide whether you would use `clickable { onClick() }` or this approach. I would recommend that you use this and omit name of parameter, since it is the same as function you are passing in. Pick whatever you like more and be consistent with it in whole project.